### PR TITLE
refactor: drop Table trait

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

This refers to #2065. 

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/table/src/thin_table.rs [hunk 1]
```diff
@@ -12,76 +12,61 @@
...
 use store_api::data_source::DataSourceRef;
 use store_api::storage::ScanRequest;
 
 use crate::error::{Result, TablesRecordBatchSnafu};
 use crate::metadata::{FilterPushDownType, TableInfoRef, TableType};
-use crate::Table;
 
 /// The `ThinTable` struct will replace the `Table` trait.
 /// TODO(zhongzc): After completion, perform renaming and documentation work.
 pub struct ThinTable {
```
Comment (tisonkun) [inline src/table/src/thin_table.rs:28]: If this PR is good to go, I can rename the struct in this PR or in a follow-up.

Comment (zhongzc) [reply src/table/src/thin_table.rs:27]: Great job! I prefer to refactor `crate::thin_table::ThinTable` to `crate::Table` in this PR.

Comment (tisonkun) [reply src/table/src/thin_table.rs]: Updated. Please take a look.

File: src/catalog/src/information_schema.rs [hunk 2]
```diff
@@ -41,7 +41,7 @@ use table::error::{SchemaConversionSnafu, TablesRecordBatchSnafu};
 use table::metadata::{
     FilterPushDownType, TableInfoBuilder, TableInfoRef, TableMetaBuilder, TableType,
 };
-use table::thin_table::{ThinTable, ThinTableAdapter};
+use table::table::Table;
 use table::TableRef;
```
Comment (zhongzc) [inline src/catalog/src/information_schema.rs:45]:
```suggestion
use table::{Table, TableRef};
```

File: src/catalog/src/information_schema.rs [hunk 3]
```diff
@@ -187,10 +187,9 @@ impl InformationSchemaProvider {
         self.information_table(name).map(|table| {
             let table_info = Self::table_info(self.catalog_name.clone(), &table);
             let filter_pushdown = FilterPushDownType::Inexact;
-            let thin_table = ThinTable::new(table_info, filter_pushdown);
-
             let data_source = Arc::new(InformationTableDataSource::new(table));
-            Arc::new(ThinTableAdapter::new(thin_table, data_source)) as _
+            let thin_table = Table::new(table_info, filter_pushdown, data_source);
+            Arc::new(thin_table)
```
Comment (zhongzc) [inline src/catalog/src/information_schema.rs:192]:
```suggestion
            let table = Table::new(table_info, filter_pushdown, data_source);
            Arc::new(table)
```

File: src/catalog/src/information_schema.rs (hunk 2 above)
Comment (tisonkun) [reply src/catalog/src/information_schema.rs]: Integrated.

File: src/query/src/tests/time_range_filter_test.rs [hunk 4]
```diff
@@ -30,8 +30,8 @@ use store_api::data_source::{DataSource, DataSourceRef};
 use store_api::storage::ScanRequest;
 use table::metadata::FilterPushDownType;
 use table::predicate::TimeRangePredicateBuilder;
+use table::table::Table;
 use table::test_util::MemTable;
-use table::thin_table::{ThinTable, ThinTableAdapter};
 use table::TableRef;
```
Comment (zhongzc) [inline src/query/src/tests/time_range_filter_test.rs:35]: ditto

File: src/query/src/tests/time_range_filter_test.rs [hunk 5]
```diff
@@ -42,16 +42,13 @@ struct MemTableWrapper;
...
-        let thin_table = ThinTable::new(table_info, FilterPushDownType::Exact);
+        let data_source = table.data_source();
         let data_source = Arc::new(DataSourceWrapper {
             inner: data_source,
             filter,
         });
-
-        Arc::new(ThinTableAdapter::new(thin_table, data_source))
+        let thin_table = Table::new(table_info, FilterPushDownType::Exact, data_source);
+        Arc::new(thin_table)
```
Comment (zhongzc) [inline src/query/src/tests/time_range_filter_test.rs:51]: ditto

File: src/table/src/dist_table.rs [hunk 6]
```diff
@@ -21,17 +21,17 @@ use store_api::storage::ScanRequest;
 
 use crate::error::UnsupportedSnafu;
 use crate::metadata::{FilterPushDownType, TableInfoRef};
-use crate::thin_table::{ThinTable, ThinTableAdapter};
+use crate::table::Table;
 use crate::TableRef;
```
Comment (zhongzc) [inline src/table/src/dist_table.rs:25]: ditto

File: src/table/src/dist_table.rs [hunk 7]
```diff
@@ -21,17 +21,17 @@ use store_api::storage::ScanRequest;
...
 #[derive(Clone)]
 pub struct DistTable;
 
 impl DistTable {
     pub fn table(table_info: TableInfoRef) -> TableRef {
-        let thin_table = ThinTable::new(table_info, FilterPushDownType::Inexact);
         let data_source = Arc::new(DummyDataSource);
-        Arc::new(ThinTableAdapter::new(thin_table, data_source))
+        let thin_table = Table::new(table_info, FilterPushDownType::Inexact, data_source);
+        Arc::new(thin_table)
```
Comment (zhongzc) [inline src/table/src/dist_table.rs:34]: ditto

File: src/table/src/table/numbers.rs [hunk 8]
```diff
@@ -32,7 +32,7 @@ use store_api::storage::ScanRequest;
 use crate::metadata::{
     FilterPushDownType, TableId, TableInfoBuilder, TableInfoRef, TableMetaBuilder, TableType,
 };
-use crate::thin_table::{ThinTable, ThinTableAdapter};
+use crate::table::Table;
 use crate::TableRef;
```
Comment (zhongzc) [inline src/table/src/table/numbers.rs:36]: ditto

File: src/table/src/table/numbers.rs [hunk 9]
```diff
@@ -49,12 +49,13 @@ impl NumbersTable {
...
-        let thin_table = ThinTable::new(
+        let data_source = Arc::new(NumbersDataSource::new(Self::schema()));
+        let thin_table = Table::new(
             Self::table_info(table_id, name, "test_engine".to_string()),
             FilterPushDownType::Unsupported,
+            data_source,
         );
-        let data_source = Arc::new(NumbersDataSource::new(Self::schema()));
-        Arc::new(ThinTableAdapter::new(thin_table, data_source))
+        Arc::new(thin_table)
```
Comment (zhongzc) [inline src/table/src/table/numbers.rs:58]: ditto

File: src/table/src/test_util/empty_table.rs [hunk 10]
```diff
@@ -21,18 +21,22 @@ use store_api::data_source::DataSource;
 use store_api::storage::ScanRequest;
 
 use crate::metadata::{FilterPushDownType, TableInfo};
-use crate::thin_table::{ThinTable, ThinTableAdapter};
+use crate::table::Table;
 use crate::TableRef;
```
Comment (zhongzc) [inline src/table/src/test_util/empty_table.rs:25]: ditto

File: src/table/src/test_util/empty_table.rs [hunk 11]
```diff
@@ -21,18 +21,22 @@ use store_api::data_source::DataSource;
...
         let data_source = Arc::new(EmptyDataSource {
             schema: info.meta.schema.clone(),
         });
-        Arc::new(ThinTableAdapter::new(thin_table, data_source))
+        let thin_table = Table::new(
+            Arc::new(info.clone()),
+            FilterPushDownType::Unsupported,
+            data_source,
+        );
+        Arc::new(thin_table)
```
Comment (zhongzc) [inline src/table/src/test_util/empty_table.rs:39]: ditto

File: src/table/src/test_util/memtable.rs [hunk 12]
```diff
@@ -33,7 +33,7 @@ use crate::error::{SchemaConversionSnafu, TableProjectionSnafu, TablesRecordBatc
 use crate::metadata::{
     FilterPushDownType, TableId, TableInfoBuilder, TableMetaBuilder, TableType, TableVersion,
 };
-use crate::thin_table::{ThinTable, ThinTableAdapter};
+use crate::table::Table;
 use crate::TableRef;
```
Comment (zhongzc) [inline src/table/src/test_util/memtable.rs:37]: ditto

File: src/table/src/test_util/memtable.rs [hunk 13]
```diff
@@ -94,9 +94,9 @@ impl MemTable {
                 .unwrap(),
         );
 
-        let thin_table = ThinTable::new(info, FilterPushDownType::Unsupported);
         let data_source = Arc::new(MemtableDataSource { recordbatch });
-        Arc::new(ThinTableAdapter::new(thin_table, data_source))
+        let thin_table = Table::new(info, FilterPushDownType::Unsupported, data_source);
+        Arc::new(thin_table)
```
Comment (zhongzc) [inline src/table/src/test_util/memtable.rs:99]: ditto

**Review by zhongzc — COMMENTED** [review]
🫣

**Review by zhongzc — APPROVED** [review]

File: src/query/src/tests/time_range_filter_test.rs (hunk 5 above)
Comment (tisonkun) [reply src/query/src/tests/time_range_filter_test.rs:51]:
```suggestion
        let table = Table::new(table_info, FilterPushDownType::Exact, data_source);
        Arc::new(table)
```

**Comment by tisonkun** [comment]
All handled.

**Review by zhongzc — APPROVED** [review]

**Review by waynexia — APPROVED** [review]

**Pull request merged**